common/cmake/devices.cmake

```diff
@@ -929,6 +929,164 @@ function(ADD_FPGA_TARGET_BOARDS)
+  # -------------------------------------------------------------------------
+  set(ALL_OUT_FASM "")
+  if(${USE_FASM})
+    foreach(TARGET ${INCLUDED_TARGETS})
+      get_target_property_required(BOARD ${TARGET} BOARD)
+      get_target_property_required(DEVICE ${BOARD} DEVICE)
+      get_target_property_required(DEVICE_TYPE ${DEVICE} DEVICE_TYPE)
+      get_target_property(USE_OVERLAY ${DEVICE_TYPE} USE_OVERLAY)
+      get_target_property_required(OUT_FASM ${TARGET} OUT_FASM)
+      if ("${FASM_TO_BIT_DEPS}" STREQUAL "FASM_TO_BIT_DEPS-NOTFOUND")
+        set(FASM_TO_BIT_DEPS "")
+      endif()
+
+      append_file_dependency(ALL_OUT_FASM ${OUT_FASM})
+
+      if(USE_OVERLAY)
+        if (NOT OVERLAY)
+          set(OVERLAY ${TARGET})
+        else()
+          message(FATAL_ERROR "More than one overlay device for ${BOARD}")
+        endif()
+      endif()
+    endforeach()
+
+    if (NOT OVERLAY)
+      list(LENGTH ${INCLUDED_TARGETS} TARGETS_LENGTH)
+      if(NOT ${TARGETS_LENGTH} EQUAL 0)
+        message(FATAL_ERROR "Multiple devices but no overlay for ${BOARD}")
+      endif()
+      set(OVERLAY ${INCLUDED_TARGETS})
+    endif()
+
+    get_target_property_required(OVERLAY_BOARD ${OVERLAY} BOARD)
+    get_target_property_required(OVERLAY_DEVICE ${OVERLAY_BOARD} DEVICE)
+    get_target_property_required(PACKAGE ${OVERLAY_BOARD} PACKAGE)
+    get_target_property_required(PYTHON3 env PYTHON3)
+
+    get_target_property(FASM_TO_BIT_EXTRA_ARGS ${OVERLAY_BOARD} FASM_TO_BIT_EXTRA_ARGS)
+    if ("${FASM_TO_BIT_EXTRA_ARGS}" STREQUAL "FASM_TO_BIT_EXTRA_ARGS-NOTFOUND")
+      set(FASM_TO_BIT_EXTRA_ARGS "")
+    endif()
+
+    get_target_property_required(OUT_FASM ${OVERLAY} OUT_FASM)
+    get_target_property_required(OUT_FASM_MERGED ${OVERLAY} OUT_FASM_MERGED)
+    get_target_property_required(OUT_LOCAL ${OVERLAY} OUT_LOCAL)
+    get_target_property_required(TOP ${OVERLAY} TOP)
+    get_target_property_required(ARCH ${OVERLAY_DEVICE} ARCH)
+    get_target_property_required(BITSTREAM_EXTENSION ${ARCH} BITSTREAM_EXTENSION)
+    get_target_property_required(FASM_TO_BIT ${ARCH} FASM_TO_BIT)
+    get_target_property_required(FASM_TO_BIT_CMD ${ARCH} FASM_TO_BIT_CMD)
+    get_target_property(FASM_TO_BIT_DEPS ${ARCH} FASM_TO_BIT_DEPS)
+    set(OUT_BITSTREAM ${OUT_LOCAL}/${TOP}.${BITSTREAM_EXTENSION})
+
+    set(BITSTREAM_DEPS ${ALL_OUT_FASM} ${FASM_TO_BIT} ${FASM_TO_BIT_DEPS})
+
+    set(OUT_FASM ${OUT_FASM_MERGED})
+    string(CONFIGURE ${FASM_TO_BIT_CMD} FASM_TO_BIT_CMD_FOR_TARGET)
+    separate_arguments(
+      FASM_TO_BIT_CMD_FOR_TARGET_LIST UNIX_COMMAND ${FASM_TO_BIT_CMD_FOR_TARGET}
+    )
+
+    separate_arguments(
+      FASM_TO_BIT_EXTRA_ARGS_LIST UNIX_COMMAND ${FASM_TO_BIT_EXTRA_ARGS}
+    )
+    set(FASM_TO_BIT_CMD_FOR_TARGET_LIST ${FASM_TO_BIT_CMD_FOR_TARGET_LIST} ${FASM_TO_BIT_EXTRA_ARGS_LIST})
+
+    add_custom_command(
+      OUTPUT ${OUT_BITSTREAM}
+      DEPENDS ${BITSTREAM_DEPS}
+      COMMAND cat ${ALL_OUT_FASM} >> ${OUT_FASM_MERGED}
```

Comment:
`ALL_OUT_FASM` is no long the list of files, it the list of dependencies.  `append_file_dependency` is for creating the dependency list, not the file location itself.  `append_file_location` should be used for creating the list of actual file locations, `append_file_dependency` is for building the dependency list